### PR TITLE
[wip] build: do not run ci for changes to some dirs

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

The success of changes to `docs/`, `script/` and `tools/` would never be reflected in debug build output from appveyor, so per [these settings](https://www.appveyor.com/docs/how-to/filtering-commits/#commit-files-github-and-bitbucket-only), configure appveyor not to run if all of the files modified in push’s head commit are in any of these folders.

This PR also adds a `preparecommitmsg` hook on CircleCI that accomplishes the same goal, as circle does not have the same convenient `skip_commit` option.

cc @jkleinsc @MarshallOfSound 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: none

### CODE REVIEW

**Comment by jkleinsc** [comment]
@codebytere Appveyor also supports CI skip in the commit msg: https://www.appveyor.com/docs/how-to/filtering-commits/#skip-commits

**Comment by codebytere** [comment]
oh yes i saw that; i was just trying to reduce the overall voodoo haha

**Comment by jkleinsc** [comment]
Yeah, I guess it doesn't hurt to be explicit in the case of AppVeyor.

File: script/check-skip-circle-ci [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+# Load in every file that will be changed via this commit into an array
+changes=( `git diff --name-only --cached` )
+commit_file="../.git/COMMIT_EDITMSG"
```
Comment (alexeykuzmin) [inline script/check-skip-circle-ci]: Isn't the file path supposed to be passed as a script argument?

File: script/check-skip-circle-ci [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+		break
+	fi
+done
+
+if [[ ${#changes[@]} -gt 0 ]]; then
+	exit
+fi
+
+# Prefix the commit message with "[skip ci]"
+sed -i.bak -e "1s/^/[skip ci] /" "$GIT_DIR/COMMIT_EDITMSG"
```
Comment (jkleinsc) [inline script/check-skip-circle-ci]: uh... what about Windows commits?

File: script/check-skip-circle-ci [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+for i in "${blacklist[@]}"
+do
+	# Remove the current pattern from the list of changes
+	changes=( ${changes[@]/$i/} )
+
+	if [[ ${#changes[@]} -eq 0 ]]; then
+		break
+	fi
+done
```
Comment (ckerr) [inline script/check-skip-circle-ci]: This seems like a lot of work. I think lines 4-18 could be replaced with this:

```sh
changes=( `git diff --name-only --cached ':!docs' ':!script' ':!tools'` )
```

**Comment by jkleinsc** [comment]
@codebytere I tested this in Windows and it doesn't appear to work:
```
PS C:\Users\jkleinsc\work\electron-gn\src\electron> git add .\docs\README.md

PS C:\Users\jkleinsc\work\electron-gn\src\electron> git commit -m "does this work"
git : husky > npm run -s precommit (node v10.9.0)
At line:1 char:1
+ git commit -m "does this work"
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (husky > npm run... (node v10.9.0):String) [], Remot 
   eException
    + FullyQualifiedErrorId : NativeCommandError
 
Stashing changes... [started]
Stashing changes... [skipped]
â†’ No partially staged files found...
Running linters... [started]
Running tasks for *.{js,ts} [started]
Running tasks for *.{cc,mm,c,h} [started]
Running tasks for *.md [started]
Running tasks for *.{gn,gni} [started]
Running tasks for *.py [started]
Running tasks for docs/api/*.md [started]
Running tasks for *.{js,ts} [skipped]
â†’ No staged files match *.{js,ts}
Running tasks for *.{cc,mm,c,h} [skipped]
â†’ No staged files match *.{cc,mm,c,h}
Running tasks for *.{gn,gni} [skipped]
â†’ No staged files match *.{gn,gni}
Running tasks for *.py [skipped]
â†’ No staged files match *.py
Running tasks for docs/api/*.md [skipped]
â†’ No staged files match docs/api/*.md
remark -qf [started]
remark -qf [completed]
Running tasks for *.md [completed]
Running linters... [completed]
husky > npm run -s preparecommitmsg (node v10.9.0)
'script' is not recognized as an internal or external command,
operable program or batch file.
husky > prepare-commit-msg hook failed (cannot be bypassed with --no-verify due to Git specs)
```

File: script/check-skip-circle-ci [hunk 4]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+# Get all changes that are not in omitted folders
+changes=( `git diff --name-only --cached ':!docs' ':!script' ':!tools'` )
+
+if [[ ${#changes[@]} -gt 0 ]]; then
+	exit
+fi
```
Comment (ckerr) [inline script/check-skip-circle-ci]: I just learned of `git --exit-code` and thought of this patch. That would be cleaner still:

```sh
if git diff --cached --no-patch --exit-code ':!docs' ':!script' ':!tools'; then
    exit
fi
```

**Comment by codebytere** [comment]
Closing for now due to cross-platform challenges and viability.

**Pull request closed without merging**